app/controllers/blazer/queries_controller.rb

```diff
@@ -3,17 +3,24 @@ class QueriesController < BaseController
     before_action :set_query, only: [:show, :edit, :update, :destroy, :refresh]
 
     def home
-      if params[:filter] == "dashboards"
-        @queries = []
-      else
+      if current_user.can?(:access, :blazer_engineering) && params[:filter] != "dashboards"
         set_queries(1000)
+      else
+        @queries = []
       end
 
       if params[:filter] && params[:filter] != "dashboards"
         @dashboards = [] # TODO show my dashboards
       else
         @dashboards = Blazer::Dashboard.order(:name)
         @dashboards = @dashboards.includes(:creator) if Blazer.user_class
+
+        # Filters dashboards to only display those the user has access to
+        @dashboards =
```

Comment:
this looks weird, not sure whats going on here or if its fully inteded

Reply:
The main page normally displays all dashboards created. Even though users only have access to those dashboards we've given them access to, @DuyH also requested that users also only be able to see links to dashboards they can access, which is what this does. Some of the weirdness is due to trying to hide the names of our roles/abilities in this repo and keep that information in our own repo, which was also requested.